packages/storycrawler/src/browser/stories-browser.ts

```diff
@@ -62,41 +72,67 @@ export class StoriesBrowser extends BaseBrowser {
         waitUntil: 'domcontentloaded',
       },
     );
-    await this.page.waitForFunction(() => (window as ExposedWindow).__STORYBOOK_CLIENT_API__, {
-      timeout: 60_000,
-    });
+    await this.page.waitForFunction(
+      () =>
+        (window as ExposedWindow).__STORYBOOK_CLIENT_API__ ||
+        (window as ExposedWindow).__STORYBOOK_PREVIEW__?.storyStoreValue,
+      {
+        timeout: 60_000,
+      },
+    );
     await this.page.evaluate(() => {
-      const { __STORYBOOK_CLIENT_API__: api } = window as ExposedWindow;
-      api.storyStore && api.storyStore.cacheAllCSFFiles();
+      const api = (window as ExposedWindow).__STORYBOOK_CLIENT_API__ || (window as ExposedWindow).__STORYBOOK_PREVIEW__;
+      function isPreviewApi(api: API | PreviewAPI): api is PreviewAPI {
+        return (api as PreviewAPI).storyStoreValue !== undefined;
+      }
+      if (api === undefined) return;
+
+      if (isPreviewApi(api)) {
+        api.storyStoreValue && api.storyStoreValue.cacheAllCSFFiles();
+        return;
+      }
+      api.storyStore?.cacheAllCSFFiles && api.storyStore.cacheAllCSFFiles();
     });
-    const result = await this.page.evaluate(
-      () =>
-        new Promise<{ stories: Story[] | null; timeout: boolean }>(res => {
-          const getStories = (count = 0) => {
-            const MAX_CONFIGURE_WAIT_COUNT = 4_000;
-            const { __STORYBOOK_CLIENT_API__: api } = window as ExposedWindow;
-            if (api.raw) {
-              // for Storybook v6
-              const configuring = api.store && api.store()._configuring;
-              // for Storybook v6 and 'storyStoreV7' option
-              const configuringV7store = api.storyStore && !api.storyStore.cachedCSFFiles;
+    const result = await this.page.evaluate(() => {
+      function isPreviewApi(api: API | PreviewAPI): api is PreviewAPI {
+        return (api as PreviewAPI).storyStoreValue !== undefined;
+      }
 
-              if (configuring || configuringV7store) {
-                if (count < MAX_CONFIGURE_WAIT_COUNT) {
-                  setTimeout(() => getStories(++count), 16);
-                } else {
-                  res({ stories: null, timeout: true });
-                }
-                return;
-              }
-              // for Storybook v5
-              const stories = api.raw().map(_ => ({ id: _.id, kind: _.kind, story: _.name, version: 'v5' } as Story));
```

Comment:
Version might have been always pointing to v5?
Since version 8 could be distinguishable, I added a change to make it either v5 or v8.

Reply:
> Version might have been always pointing to v5?

Yes. The `version` field comes from the fact that there was a significant difference in the way to manipulate SB's preview iframe operated between Storybook v4 and v5.

c.f.: https://github.com/reg-viz/storycap/pull/639/files#diff-c1400948b89a47066e7596974308900b8ed46e395b5649f555efc69fab7cb32d

If there is no change in the way to operate the preview iframe in Storybook v8, then I think we can keep the `version` field v5.

Reply:
Since there seems to be no change in the way to operate the preview iframe in version 8, i brought back the v5 with [456f8eb](https://github.com/reg-viz/storycap/pull/855/commits/456f8eb28dd1df3a7998cf71aae260493f81f623)